src/utils/validators.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  IResponseErrorValidation,
+  ResponseErrorValidation
+} from "@pagopa/ts-commons/lib/responses";
+import { pipe } from "fp-ts/lib/function";
+
+import * as express from "express";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { FiscalCode } from "../generated/FiscalCode";
+import { validIunList, validLegalFactList } from "../__mocks__/variables";
+import { LegalFactType } from "./types";
+import { IPNResponseErrorProblem, PNResponseErrorProblem } from "./responses";
+
+//this function should be replaced with the nearly next one, i'm not removing this at the moment to not break anything
+export const validateTaxIdHeader = (req: express.Request) => (): TE.TaskEither<
+  IResponseErrorValidation,
+  FiscalCode
+> =>
+  pipe(
+    req.headers["x-pagopa-cx-taxid"],
+    FiscalCode.decode,
+    E.mapLeft(() =>
+      ResponseErrorValidation(
+        "Error while processing request",
+        "Missing x-pagopa-cx-taxid header"
+      )
+    ),
+    TE.fromEither
+  );
+
+export const validateTaxIdInHeader = (
+  req: express.Request
+) => (): TE.TaskEither<IPNResponseErrorProblem, FiscalCode> =>
```

Comment:
this doesn't need to be a curried function based on the use you are doing with it